server.go

```diff
@@ -982,9 +984,32 @@ func (s *server) sendPeerMessages(
 		defer wg.Done()
 	}
 
+	// We queue each message, creating a slice of error channels that
+	// can be inspected after every message is successfully added to
+	// the queue.
+	var errChans []chan error
 	for _, msg := range msgs {
-		targetPeer.queueMsg(msg, nil)
+		errChan := make(chan error, 1)
+		if err := targetPeer.queueMsg(msg, errChan); err != nil {
```

Comment:
If we modify `queueMsg` to just send an error, or return one (not both), then we don't also need to check the error here. As otherwise, it may _always_ block.